source/common/ssl/context_impl.cc

```diff
@@ -47,8 +49,25 @@ ContextImpl::ContextImpl(ContextManagerImpl& parent, Stats::Scope& scope, Contex
     rc = SSL_CTX_add_client_CA(ctx_.get(), ca_cert_.get());
     RELEASE_ASSERT(1 == rc);
 
-    // enable peer certificate verification
-    SSL_CTX_set_verify(ctx_.get(), SSL_VERIFY_PEER, nullptr);
+    verify_mode = SSL_VERIFY_PEER;
+  }
+
+  if (!config.verifySubjectAltNameList().empty()) {
+    verify_subject_alt_name_list_ = config.verifySubjectAltNameList();
+    verify_mode = SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT;
```

Comment:
I believe that we'll be using SANs in client certificates for service mesh authentication with [SPIFFE](https://spiffe.io/).